samples/extensions/descriptor_buffer_basic/descriptor_buffer_basic.cpp

```diff
@@ -0,0 +1,471 @@
+
+		// Global Matrices (set 0)
+		vkCmdSetDescriptorBufferOffsetsEXT(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 0, 1, &buffer_index_ubo, &buffer_offset);
+
+		// Set and offset into descriptor for each model
+		for (uint32_t j = 0; j < static_cast<uint32_t>(cubes.size()); j++)
+		{
+			// Uniform buffer (set 1)
+			// Model ubos start at offset * 1 (slot 0 is global matrices)
+			buffer_offset = alignment + j * alignment;
+			vkCmdSetDescriptorBufferOffsetsEXT(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 1, 1, &buffer_index_ubo, &buffer_offset);
+			// Image (set 2)
+			uint32_t buffer_index_image = 1;
+			buffer_offset               = j * alignment;
+			vkCmdSetDescriptorBufferOffsetsEXT(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 2, 1, &buffer_index_image, &buffer_offset);
+			draw_model(models.cube, draw_cmd_buffers[i]);
+		}
+
+		// @todo: can't mix descriptor buffers and descriptors, so we disable the UI for now
+		// draw_ui(draw_cmd_buffers[i]);
+
+		vkCmdEndRenderPass(draw_cmd_buffers[i]);
+
+		VK_CHECK(vkEndCommandBuffer(draw_cmd_buffers[i]));
+	}
+}
+
+void DescriptorBufferBasic::load_assets()
+{
+	models.cube      = load_model("scenes/textured_unit_cube.gltf");
+	cubes[0].texture = load_texture("textures/crate01_color_height_rgba.ktx", vkb::sg::Image::Color);
+	cubes[1].texture = load_texture("textures/crate02_color_height_rgba.ktx", vkb::sg::Image::Color);
+}
+
+void DescriptorBufferBasic::setup_descriptor_set_layout()
+{
+	// Using descriptor buffers still requires use to create descriptor set layouts
+
+	VkDescriptorSetLayoutBinding set_layout_binding{};
+
+	VkDescriptorSetLayoutCreateInfo descriptor_layout_create_info{};
+	descriptor_layout_create_info.sType        = VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_CREATE_INFO;
+	descriptor_layout_create_info.bindingCount = 1;
+	descriptor_layout_create_info.flags        = VK_DESCRIPTOR_SET_LAYOUT_CREATE_DESCRIPTOR_BUFFER_BIT_EXT;
+	descriptor_layout_create_info.pBindings    = &set_layout_binding;
+
+	// Create a layout for uniform buffers
+	set_layout_binding = vkb::initializers::descriptor_set_layout_binding(VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, VK_SHADER_STAGE_VERTEX_BIT, 0);
+	VK_CHECK(vkCreateDescriptorSetLayout(get_device().get_handle(), &descriptor_layout_create_info, nullptr, &descriptor_set_layout_buffer));
+
+	// Create a layout for combined image samplers
+	set_layout_binding = vkb::initializers::descriptor_set_layout_binding(VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, VK_SHADER_STAGE_FRAGMENT_BIT, 0);
+	VK_CHECK(vkCreateDescriptorSetLayout(get_device().get_handle(), &descriptor_layout_create_info, nullptr, &descriptor_set_layout_image));
+
+	// Create a pipeline layout using set 0 = Camera UBO, set 1 = Model UBO and set 2 = Model combined image
+	const std::array<VkDescriptorSetLayout, 3> decriptor_set_layouts = {descriptor_set_layout_buffer, descriptor_set_layout_buffer, descriptor_set_layout_image};
```

Comment:
rename the variable decriptor_set_layouts to descriptor_set_layouts.  (descriptor is missing the 's')